### PR TITLE
Add new linters

### PR DESCRIPTION
**Pull request by johejo**
## Description

Enable some linters and fix issues.
Fixed some deprecated and dangerous code.

I chose linter carefully, but would like to respect the opinions of the maintainers, so please let me know thoughts.

## Motivation and Context

We want to make incremental improvements to maintain our code in a sustainable way.
This is the first step.

## How Has This Been Tested?

`make test`

## Checklist:

- [ ] My change requires a change to the documentation or CHANGELOG.
- [x] I have updated the documentation/CHANGELOG accordingly.
- [x] I have created a feature (non-master) branch for my PR.

### CODE REVIEW

File: oauthproxy.go [hunk 1]
```diff
@@ -324,8 +324,7 @@ func (p *OAuthProxy) GetRedirectURI(host string) string {
 	if p.redirectURL.Host != "" {
 		return p.redirectURL.String()
 	}
-	var u url.URL
-	u = *p.redirectURL
+	var u url.URL = *p.redirectURL
```
Comment (JoelSpeed) [inline oauthproxy.go:327]:
```suggestion
	u := *p.redirectURL
```

File: oauthproxy.go [hunk 2]
```diff
@@ -1096,10 +1095,7 @@ func (p *OAuthProxy) CheckBasicAuth(req *http.Request) (*sessionsapi.SessionStat
 
 // isAjax checks if a request is an ajax request
 func isAjax(req *http.Request) bool {
-	acceptValues, ok := req.Header["accept"]
-	if !ok {
-		acceptValues = req.Header["Accept"]
-	}
+	acceptValues := req.Header.Values("Accept")
```
Comment (JoelSpeed) [inline oauthproxy.go:1098]: Is this change functionally equivalent? ie are the headers case sensitive?

File: providers/logingov.go [hunk 3]
```diff
@@ -260,8 +260,7 @@ func (p *LoginGovProvider) Redeem(redirectURL, code string) (s *sessions.Session
 
 // GetLoginURL overrides GetLoginURL to add login.gov parameters
 func (p *LoginGovProvider) GetLoginURL(redirectURI, state string) string {
-	var a url.URL
-	a = *p.LoginURL
+	var a url.URL = *p.LoginURL
```
Comment (JoelSpeed) [inline providers/logingov.go:263]:
```suggestion
	a := *p.LoginURL
```

File: providers/provider_default.go [hunk 4]
```diff
@@ -86,8 +86,7 @@ func (p *ProviderData) Redeem(redirectURL, code string) (s *sessions.SessionStat
 
 // GetLoginURL with typical oauth parameters
 func (p *ProviderData) GetLoginURL(redirectURI, state string) string {
-	var a url.URL
-	a = *p.LoginURL
+	var a url.URL = *p.LoginURL
```
Comment (JoelSpeed) [inline providers/provider_default.go:89]:
```suggestion
	a := *p.LoginURL
```

File: .golangci.yml [hunk 5]
```diff
@@ -9,5 +9,28 @@ linters:
...
+    - gosimple
+    - staticcheck
+    - structcheck
+    - typecheck
+    - unused
+    - varcheck
+    - bodyclose
+    - dogsled
+    - goprintffuncname
+    - interfacer
```
Comment (JoelSpeed) [inline .golangci.yml:21]: I know interfacer is deprecated so please remove this one

File: oauthproxy.go (hunk 2 above)
Comment (johejo) [reply oauthproxy.go:1098]: The `Values` (since Go1.14) returns headers that are case insensitive.
https://golang.org/pkg/net/http/#Header.Values

**Pull request merged**